pallets/nominee-election/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright 2021 Parallel Finance Developer.
+// This file is part of Parallel Finance.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Prices pallet
+//!
+//! ## Overview
+//!
+//! This pallet stores the offchain elected validators on-chain and maintain
+//! a whitelisted validators which can be selected by council.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+use frame_support::{pallet_prelude::*, traits::SortedMembers, transactional};
+use frame_system::pallet_prelude::*;
+pub use pallet::*;
+
+use sp_std::{convert::TryInto, vec::Vec};
+
+/// Nominee Election Coefficients
+/// https://docs.parallel.fi/dev/staking/staking-election
+#[derive(Encode, Decode, Eq, PartialEq, Clone, RuntimeDebug, Default)]
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+pub struct NomineeScoreCoefficients {
```

Comment:
Do we need to record this parameter now?
probably only record the validator id and score is enough, if the score calculation become onchain, it can be furthur added.

Reply:
this is for governance